web-ui/view/step.ml

```diff
@@ -116,9 +116,11 @@ module Make (M : Git_forge_intf.Forge) = struct
       :: Common.rebuild_button ~hash ~csrf_token ~show:show_rebuild ()
     in
     let title_card =
+      let ref_path = Result.get_ok (M.ref_path (List.hd refs)) in
```

Comment:
`refs` is the list of refs that have this commit as their *head*, for non-head commits clicking on them in the history page causes a 500 error due to the `List.hd`.

This is due to the step page having the 'build history' button. We could either discriminate between head commits *with* the button and historical commits *without*, or somehow pass the correct refs to historical commits.

Reply:
Considering our time, I suggest removing the `history` button for the `history` commit. Based on the suggestion, I tried [this ugly piece](https://github.com/maiste/ocaml-ci/commit/95bc6749f7c5c1ad3efa10dc63ac2acc0c7f6a20) of code, and it's working as expected. IMHO, this is the best solution. Also, we should remove the rebuild button for this solution for now.

Reply:
Thanks. That piece of code also came to me in the middle of the night ... must be the right solution ;)